File: worker-python/src/models/Organization.py
from datetime import datetime
from typing import Optional
from beanie import Document
from pydantic import Field

class Organization(Document):
    github_org_id: int = Field(..., alias="githubOrgId")
    name: str
    slug: str
    avatar_url: Optional[str] = Field(None, alias="avatarUrl")
    installation_id: Optional[int] = Field(None, alias="installationId")
    installation_status: str = Field("active", alias="installationStatus")
    settings: dict = Field(default_factory=dict)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "organizations"
        indexes = [
            "githubOrgId",
            "slug",
        ]

File: api-node/src/models/Organization.ts
import { Schema, model, Document } from "mongoose";

export interface IOrganization extends Document {
  githubOrgId: number;
  name: string;
  slug: string;
  avatarUrl?: string;
  installationId?: number;
  installationStatus: "pending" | "active" | "suspended" | "deleted";
  settings: Record<string, any>;
  createdAt: Date;
  updatedAt: Date;
}

const organizationSchema = new Schema<IOrganization>(
  {
    githubOrgId: { type: Number, required: true, unique: true, index: true },
    name: { type: String, required: true },
    slug: { type: String, required: true, unique: true, index: true },
    avatarUrl: { type: String },
    installationId: { type: Number, index: true },
    installationStatus: {
      type: String,
      enum: ["pending", "active", "suspended", "deleted"],
      default: "pending",
    },
    settings: { type: Schema.Types.Mixed, default: {} },
  },
  {
    timestamps: true,
  }
);

export const Organization = model<IOrganization>(
  "Organization",
  organizationSchema
);
